File: binding/src/libqscreen.cpp
#include <QObject>
#include <QPixmap>
#include <QPoint>
#include <QRect>
#include <QScreen>
#include <QSize>
#include <QSizeF>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <type_traits>
#include <QTransform>
#include <qscreen.h>
#include "libqscreen.h"
#include "libqscreen.hxx"

libqt_string QScreen_Tr(const char* s) {
	QString _ret = QScreen::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QScreen_Name(const QScreen* self) {
	QString _ret = self->name();
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QScreen_Manufacturer(const QScreen* self) {
	QString _ret = self->manufacturer();
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QScreen_Model(const QScreen* self) {
	QString _ret = self->model();
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QScreen_SerialNumber(const QScreen* self) {
	QString _ret = self->serialNumber();
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

int QScreen_Depth(const QScreen* self) {
	return self->depth();
}

QSize* QScreen_Size(const QScreen* self) {
	return new QSize(self->size());
}

QRect* QScreen_Geometry(const QScreen* self) {
	return new QRect(self->geometry());
}

QSizeF* QScreen_PhysicalSize(const QScreen* self) {
	return new QSizeF(self->physicalSize());
}

double QScreen_PhysicalDotsPerInchX(const QScreen* self) {
	return self->physicalDotsPerInchX();
}

double QScreen_PhysicalDotsPerInchY(const QScreen* self) {
	return self->physicalDotsPerInchY();
}

double QScreen_PhysicalDotsPerInch(const QScreen* self) {
	return self->physicalDotsPerInch();
}

double QScreen_LogicalDotsPerInchX(const QScreen* self) {
	return self->logicalDotsPerInchX();
}

double QScreen_LogicalDotsPerInchY(const QScreen* self) {
	return self->logicalDotsPerInchY();
}

double QScreen_LogicalDotsPerInch(const QScreen* self) {
	return self->logicalDotsPerInch();
}

double QScreen_DevicePixelRatio(const QScreen* self) {
	return self->devicePixelRatio();
}

QSize* QScreen_AvailableSize(const QScreen* self) {
	return new QSize(self->availableSize());
}

QRect* QScreen_AvailableGeometry(const QScreen* self) {
	return new QRect(self->availableGeometry());
}

libqt_list QScreen_VirtualSiblings(const QScreen* self) {
	auto _ret = self->virtualSiblings();
	libqt_list _arr;
	_arr.len = _ret.length();
	_arr.data = malloc(_arr.len * sizeof(void*));
	void** _data = static_cast<void**>(_arr.data);
	for (int _i = 0; _i < _arr.len; ++_i) {
		_data[_i] = _ret[_i];
	}
	return _arr;
}

QScreen* QScreen_VirtualSiblingAt(QScreen* self, QPoint* point) {
	return self->virtualSiblingAt(*point);
}

QSize* QScreen_VirtualSize(const QScreen* self) {
	return new QSize(self->virtualSize());
}

QRect* QScreen_VirtualGeometry(const QScreen* self) {
	return new QRect(self->virtualGeometry());
}

QSize* QScreen_AvailableVirtualSize(const QScreen* self) {
	return new QSize(self->availableVirtualSize());
}

QRect* QScreen_AvailableVirtualGeometry(const QScreen* self) {
	return new QRect(self->availableVirtualGeometry());
}

int QScreen_PrimaryOrientation(const QScreen* self) {
	return self->primaryOrientation();
}

int QScreen_Orientation(const QScreen* self) {
	return self->orientation();
}

int QScreen_NativeOrientation(const QScreen* self) {
	return self->nativeOrientation();
}

int QScreen_AngleBetween(const QScreen* self, int a, int b) {
	return self->angleBetween(static_cast<Qt::ScreenOrientation>(a), static_cast<Qt::ScreenOrientation>(b));
}

QTransform* QScreen_TransformBetween(const QScreen* self, int a, int b, const QRect* target) {
	return new QTransform(self->transformBetween(static_cast<Qt::ScreenOrientation>(a), static_cast<Qt::ScreenOrientation>(b), *target));
}

QRect* QScreen_MapBetween(const QScreen* self, int a, int b, const QRect* rect) {
	return new QRect(self->mapBetween(static_cast<Qt::ScreenOrientation>(a), static_cast<Qt::ScreenOrientation>(b), *rect));
}

bool QScreen_IsPortrait(const QScreen* self, int orientation) {
	return self->isPortrait(static_cast<Qt::ScreenOrientation>(orientation));
}

bool QScreen_IsLandscape(const QScreen* self, int orientation) {
	return self->isLandscape(static_cast<Qt::ScreenOrientation>(orientation));
}

QPixmap* QScreen_GrabWindow(QScreen* self) {
	return new QPixmap(self->grabWindow());
}

double QScreen_RefreshRate(const QScreen* self) {
	return self->refreshRate();
}

void QScreen_GeometryChanged(QScreen* self, const QRect* geometry) {
	self->geometryChanged(*geometry);
}

void QScreen_Connect_GeometryChanged(QScreen* self, intptr_t slot) {
    void (*slotFunc)(QScreen*, const QRect*) = reinterpret_cast<void (*)(QScreen*, const QRect*)>(slot);
    QScreen::connect(self, &QScreen::geometryChanged, [self, slotFunc](const QRect& geometry) {
	slotFunc(self, geometry);
    });
}

void QScreen_AvailableGeometryChanged(QScreen* self, const QRect* geometry) {
	self->availableGeometryChanged(*geometry);
}

void QScreen_Connect_AvailableGeometryChanged(QScreen* self, intptr_t slot) {
    void (*slotFunc)(QScreen*, const QRect*) = reinterpret_cast<void (*)(QScreen*, const QRect*)>(slot);
    QScreen::connect(self, &QScreen::availableGeometryChanged, [self, slotFunc](const QRect& geometry) {
	slotFunc(self, geometry);
    });
}

void QScreen_PhysicalSizeChanged(QScreen* self, const QSizeF* size) {
	self->physicalSizeChanged(*size);
}

void QScreen_Connect_PhysicalSizeChanged(QScreen* self, intptr_t slot) {
    void (*slotFunc)(QScreen*, const QSizeF*) = reinterpret_cast<void (*)(QScreen*, const QSizeF*)>(slot);
    QScreen::connect(self, &QScreen::physicalSizeChanged, [self, slotFunc](const QSizeF& size) {
	slotFunc(self, size);
    });
}

void QScreen_PhysicalDotsPerInchChanged(QScreen* self, double dpi) {
	self->physicalDotsPerInchChanged(dpi);
}

void QScreen_Connect_PhysicalDotsPerInchChanged(QScreen* self, intptr_t slot) {
    void (*slotFunc)(QScreen*, double) = reinterpret_cast<void (*)(QScreen*, double)>(slot);
    QScreen::connect(self, &QScreen::physicalDotsPerInchChanged, [self, slotFunc](qreal dpi) {
	slotFunc(self, dpi);
    });
}

void QScreen_LogicalDotsPerInchChanged(QScreen* self, double dpi) {
	self->logicalDotsPerInchChanged(dpi);
}

void QScreen_Connect_LogicalDotsPerInchChanged(QScreen* self, intptr_t slot) {
    void (*slotFunc)(QScreen*, double) = reinterpret_cast<void (*)(QScreen*, double)>(slot);
    QScreen::connect(self, &QScreen::logicalDotsPerInchChanged, [self, slotFunc](qreal dpi) {
	slotFunc(self, dpi);
    });
}

void QScreen_VirtualGeometryChanged(QScreen* self, const QRect* rect) {
	self->virtualGeometryChanged(*rect);
}

void QScreen_Connect_VirtualGeometryChanged(QScreen* self, intptr_t slot) {
    void (*slotFunc)(QScreen*, const QRect*) = reinterpret_cast<void (*)(QScreen*, const QRect*)>(slot);
    QScreen::connect(self, &QScreen::virtualGeometryChanged, [self, slotFunc](const QRect& rect) {
	slotFunc(self, rect);
    });
}

void QScreen_PrimaryOrientationChanged(QScreen* self, int orientation) {
	self->primaryOrientationChanged(static_cast<Qt::ScreenOrientation>(orientation));
}

void QScreen_Connect_PrimaryOrientationChanged(QScreen* self, intptr_t slot) {
    void (*slotFunc)(QScreen*, int) = reinterpret_cast<void (*)(QScreen*, int)>(slot);
    QScreen::connect(self, &QScreen::primaryOrientationChanged, [self, slotFunc](Qt::ScreenOrientation orientation) {
	slotFunc(self, orientation);
    });
}

void QScreen_OrientationChanged(QScreen* self, int orientation) {
	self->orientationChanged(static_cast<Qt::ScreenOrientation>(orientation));
}

void QScreen_Connect_OrientationChanged(QScreen* self, intptr_t slot) {
    void (*slotFunc)(QScreen*, int) = reinterpret_cast<void (*)(QScreen*, int)>(slot);
    QScreen::connect(self, &QScreen::orientationChanged, [self, slotFunc](Qt::ScreenOrientation orientation) {
	slotFunc(self, orientation);
    });
}

void QScreen_RefreshRateChanged(QScreen* self, double refreshRate) {
	self->refreshRateChanged(refreshRate);
}

void QScreen_Connect_RefreshRateChanged(QScreen* self, intptr_t slot) {
    void (*slotFunc)(QScreen*, double) = reinterpret_cast<void (*)(QScreen*, double)>(slot);
    QScreen::connect(self, &QScreen::refreshRateChanged, [self, slotFunc](qreal refreshRate) {
	slotFunc(self, refreshRate);
    });
}

libqt_string QScreen_Tr2(const char* s, const char* c) {
	QString _ret = QScreen::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QScreen_Tr3(const char* s, const char* c, int n) {
	QString _ret = QScreen::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

QPixmap* QScreen_GrabWindow1(QScreen* self, uintptr_t window) {
	return new QPixmap(self->grabWindow(window));
}

QPixmap* QScreen_GrabWindow2(QScreen* self, uintptr_t window, int x) {
	return new QPixmap(self->grabWindow(window, x));
}

QPixmap* QScreen_GrabWindow3(QScreen* self, uintptr_t window, int x, int y) {
	return new QPixmap(self->grabWindow(window, x, y));
}

QPixmap* QScreen_GrabWindow4(QScreen* self, uintptr_t window, int x, int y, int w) {
	return new QPixmap(self->grabWindow(window, x, y, w));
}

QPixmap* QScreen_GrabWindow5(QScreen* self, uintptr_t window, int x, int y, int w, int h) {
	return new QPixmap(self->grabWindow(window, x, y, w, h));
}

void QScreen_Delete(QScreen* self) {
    delete self;
}

File: binding/include/libqscreen.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qscreen.html
pub const qscreen = struct {

    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QScreen_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qscreen.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Name(self: ?*anyopaque, , allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QScreen_Name(@ptrCast(self));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qscreen.Name: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Manufacturer(self: ?*anyopaque, , allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QScreen_Manufacturer(@ptrCast(self));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qscreen.Manufacturer: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Model(self: ?*anyopaque, , allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QScreen_Model(@ptrCast(self));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qscreen.Model: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn SerialNumber(self: ?*anyopaque, , allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QScreen_SerialNumber(@ptrCast(self));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qscreen.SerialNumber: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Depth(self: ?*anyopaque, ) i32 {
        return qtc.QScreen_Depth(@ptrCast(self));
    }

    pub fn Size(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QScreen_Size(@ptrCast(self));
    }

    pub fn Geometry(self: ?*anyopaque, ) QtC.QRect {
        return qtc.QScreen_Geometry(@ptrCast(self));
    }

    pub fn PhysicalSize(self: ?*anyopaque, ) QtC.QSizeF {
        return qtc.QScreen_PhysicalSize(@ptrCast(self));
    }

    pub fn PhysicalDotsPerInchX(self: ?*anyopaque, ) f64 {
        return qtc.QScreen_PhysicalDotsPerInchX(@ptrCast(self));
    }

    pub fn PhysicalDotsPerInchY(self: ?*anyopaque, ) f64 {
        return qtc.QScreen_PhysicalDotsPerInchY(@ptrCast(self));
    }

    pub fn PhysicalDotsPerInch(self: ?*anyopaque, ) f64 {
        return qtc.QScreen_PhysicalDotsPerInch(@ptrCast(self));
    }

    pub fn LogicalDotsPerInchX(self: ?*anyopaque, ) f64 {
        return qtc.QScreen_LogicalDotsPerInchX(@ptrCast(self));
    }

    pub fn LogicalDotsPerInchY(self: ?*anyopaque, ) f64 {
        return qtc.QScreen_LogicalDotsPerInchY(@ptrCast(self));
    }

    pub fn LogicalDotsPerInch(self: ?*anyopaque, ) f64 {
        return qtc.QScreen_LogicalDotsPerInch(@ptrCast(self));
    }

    pub fn DevicePixelRatio(self: ?*anyopaque, ) f64 {
        return qtc.QScreen_DevicePixelRatio(@ptrCast(self));
    }

    pub fn AvailableSize(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QScreen_AvailableSize(@ptrCast(self));
    }

    pub fn AvailableGeometry(self: ?*anyopaque, ) QtC.QRect {
        return qtc.QScreen_AvailableGeometry(@ptrCast(self));
    }

    pub fn VirtualSiblings(self: ?*anyopaque, ) []const u8 {
        return qtc.QScreen_VirtualSiblings(@ptrCast(self));
    }

    pub fn VirtualSiblingAt(self: ?*anyopaque, point: QtC.QPoint) QtC.QScreen {
        return qtc.QScreen_VirtualSiblingAt(@ptrCast(self), @ptrCast(point));
    }

    pub fn VirtualSize(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QScreen_VirtualSize(@ptrCast(self));
    }

    pub fn VirtualGeometry(self: ?*anyopaque, ) QtC.QRect {
        return qtc.QScreen_VirtualGeometry(@ptrCast(self));
    }

    pub fn AvailableVirtualSize(self: ?*anyopaque, ) QtC.QSize {
        return qtc.QScreen_AvailableVirtualSize(@ptrCast(self));
    }

    pub fn AvailableVirtualGeometry(self: ?*anyopaque, ) QtC.QRect {
        return qtc.QScreen_AvailableVirtualGeometry(@ptrCast(self));
    }

    pub fn PrimaryOrientation(self: ?*anyopaque, ) i32 {
        return qtc.QScreen_PrimaryOrientation(@ptrCast(self));
    }

    pub fn Orientation(self: ?*anyopaque, ) i32 {
        return qtc.QScreen_Orientation(@ptrCast(self));
    }

    pub fn NativeOrientation(self: ?*anyopaque, ) i32 {
        return qtc.QScreen_NativeOrientation(@ptrCast(self));
    }

    pub fn AngleBetween(self: ?*anyopaque, a: i32, b: i32) i32 {
        return qtc.QScreen_AngleBetween(@ptrCast(self), @intCast(a), @intCast(b));
    }

    pub fn TransformBetween(self: ?*anyopaque, a: i32, b: i32, target: ?*anyopaque) QtC.QTransform {
        return qtc.QScreen_TransformBetween(@ptrCast(self), @intCast(a), @intCast(b), @ptrCast(target));
    }

    pub fn MapBetween(self: ?*anyopaque, a: i32, b: i32, rect: ?*anyopaque) QtC.QRect {
        return qtc.QScreen_MapBetween(@ptrCast(self), @intCast(a), @intCast(b), @ptrCast(rect));
    }

    pub fn IsPortrait(self: ?*anyopaque, orientation: i32) bool {
        return qtc.QScreen_IsPortrait(@ptrCast(self), @intCast(orientation));
    }

    pub fn IsLandscape(self: ?*anyopaque, orientation: i32) bool {
        return qtc.QScreen_IsLandscape(@ptrCast(self), @intCast(orientation));
    }

    pub fn GrabWindow(self: ?*anyopaque, ) QtC.QPixmap {
        return qtc.QScreen_GrabWindow(@ptrCast(self));
    }

    pub fn RefreshRate(self: ?*anyopaque, ) f64 {
        return qtc.QScreen_RefreshRate(@ptrCast(self));
    }

    pub fn GeometryChanged(self: ?*anyopaque, geometry: ?*anyopaque) void {
        qtc.QScreen_GeometryChanged(@ptrCast(self), @ptrCast(geometry));
    }

    pub fn OnGeometryChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QScreen_Connect_GeometryChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn AvailableGeometryChanged(self: ?*anyopaque, geometry: ?*anyopaque) void {
        qtc.QScreen_AvailableGeometryChanged(@ptrCast(self), @ptrCast(geometry));
    }

    pub fn OnAvailableGeometryChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QScreen_Connect_AvailableGeometryChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn PhysicalSizeChanged(self: ?*anyopaque, size: ?*anyopaque) void {
        qtc.QScreen_PhysicalSizeChanged(@ptrCast(self), @ptrCast(size));
    }

    pub fn OnPhysicalSizeChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QScreen_Connect_PhysicalSizeChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn PhysicalDotsPerInchChanged(self: ?*anyopaque, dpi: f64) void {
        qtc.QScreen_PhysicalDotsPerInchChanged(@ptrCast(self), dpi);
    }

    pub fn OnPhysicalDotsPerInchChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, f64) callconv(.c) void) void {
        qtc.QScreen_Connect_PhysicalDotsPerInchChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn LogicalDotsPerInchChanged(self: ?*anyopaque, dpi: f64) void {
        qtc.QScreen_LogicalDotsPerInchChanged(@ptrCast(self), dpi);
    }

    pub fn OnLogicalDotsPerInchChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, f64) callconv(.c) void) void {
        qtc.QScreen_Connect_LogicalDotsPerInchChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn VirtualGeometryChanged(self: ?*anyopaque, rect: ?*anyopaque) void {
        qtc.QScreen_VirtualGeometryChanged(@ptrCast(self), @ptrCast(rect));
    }

    pub fn OnVirtualGeometryChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QScreen_Connect_VirtualGeometryChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn PrimaryOrientationChanged(self: ?*anyopaque, orientation: i32) void {
        qtc.QScreen_PrimaryOrientationChanged(@ptrCast(self), @intCast(orientation));
    }

    pub fn OnPrimaryOrientationChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) void) void {
        qtc.QScreen_Connect_PrimaryOrientationChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn OrientationChanged(self: ?*anyopaque, orientation: i32) void {
        qtc.QScreen_OrientationChanged(@ptrCast(self), @intCast(orientation));
    }

    pub fn OnOrientationChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) void) void {
        qtc.QScreen_Connect_OrientationChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn RefreshRateChanged(self: ?*anyopaque, refreshRate: f64) void {
        qtc.QScreen_RefreshRateChanged(@ptrCast(self), refreshRate);
    }

    pub fn OnRefreshRateChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, f64) callconv(.c) void) void {
        qtc.QScreen_Connect_RefreshRateChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QScreen_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qscreen.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QScreen_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qscreen.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn GrabWindow1(self: ?*anyopaque, window: usize) QtC.QPixmap {
        return qtc.QScreen_GrabWindow1(@ptrCast(self), window);
    }

    pub fn GrabWindow2(self: ?*anyopaque, window: usize, x: i32) QtC.QPixmap {
        return qtc.QScreen_GrabWindow2(@ptrCast(self), window, x);
    }

    pub fn GrabWindow3(self: ?*anyopaque, window: usize, x: i32, y: i32) QtC.QPixmap {
        return qtc.QScreen_GrabWindow3(@ptrCast(self), window, x, y);
    }

    pub fn GrabWindow4(self: ?*anyopaque, window: usize, x: i32, y: i32, w: i32) QtC.QPixmap {
        return qtc.QScreen_GrabWindow4(@ptrCast(self), window, x, y, w);
    }

    pub fn GrabWindow5(self: ?*anyopaque, window: usize, x: i32, y: i32, w: i32, h: i32) QtC.QPixmap {
        return qtc.QScreen_GrabWindow5(@ptrCast(self), window, x, y, w, h);
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QScreen_Delete(@ptrCast(self));
    }
};
